integration-tests/tests/src/hooks/import-app-before.ts

```diff
@@ -34,20 +46,95 @@ export type AppConfigurationRelaxed = {
   baseFilePath?: string;
 };
 
+function getCredentials(): Credentials {
+  if (typeof publicKey === "string" && typeof privateKey === "string") {
+    return {
+      kind: "api-key",
+      publicKey,
+      privateKey,
+    };
+  } else {
+    return {
+      kind: "username-password",
+      username,
+      password,
+    };
+  }
+}
+
+const credentials = getCredentials();
+
+const importer = new AppImporter({
+  baseUrl,
+  credentials,
+  reuseApp,
+});
+
+function isConnectionRefused(err: unknown) {
+  return (
+    err instanceof Error &&
+    err.cause instanceof AggregateError &&
+    "code" in err.cause &&
+    err.cause.code === "ECONNREFUSED"
+  );
+}
+
+function printWarningBox(...lines: string[]) {
```

Comment:
We may want to utilize this elsewhere as well, could we move this into `/utils`? 🙂